module/vision/BallDetector/src/BallDetector.cpp

```diff
@@ -72,7 +71,7 @@ namespace vision {
                 // Partition the indices such that we only have the ball points that dont have ball surrounding them
                 auto boundary = utility::vision::visualmesh::partition_points(
                     indices.begin(), indices.end(), neighbours, [&](const int& idx) {
-                        return idx == indices.size() || (cls(BALL_INDEX, idx) >= config.confidence_threshold);
+                        return idx == (int) indices.size() || (cls(BALL_INDEX, idx) >= config.confidence_threshold);
```

Comment:
Never use c style casts, they're always the wrong choice. Depending on the circumstance you should be using either `reinterpret_cast`, `static_cast` or a functional cast `int(indices.size())`

Reply:
Which is correct in this circumstance?

Reply:
Either functional or static cast, Personally I prefer functional as it's shorter.